runtime/shutdown/shutdown.go

```diff
@@ -0,0 +1,86 @@
+package shutdown
+
+import "context"
+
+// Progress provides progress information about an ongoing graceful shutdown process.
+//
+// The process broadly consists of  phases:
+//
+// 1. Drain active handlers
+//
+// As soon as the graceful shutdown process is initiated, the service will stop accepting new
+// incoming API calls and Pub/Sub messages. It will continue to process already running handlers
+// until they complete (or the ForceCloseHandlers deadline is reached).
+//
+// This phase continues until all active handlers have completed or the ForceCloseHandlers deadline
+// is reached, whichever happens first. The ActiveEndpoints, ActivePubSubMessages, and ActiveHandlers
```

Comment:
These contexts don't exist within the type;

Did you mean `OutstandingRequests` & `OutstandingPubSubMessages`?